src/test/java/com/squareup/javapoet/CodeBlockTest.java

```diff
@@ -339,4 +339,27 @@
     CodeBlock joined = codeBlocks.stream().collect(CodeBlock.joining(" || ", "start {", "} end"));
     assertThat(joined.toString()).isEqualTo("start {\"hello\" || world.World || need tacos} end");
   }
+
+  @Test public void singleQuoteCharacterInLiteral() {
+    char ch = '\'';
+    CodeBlock chInLiteralBlock = CodeBlock.builder().add("'$L'", ch).build();
+    // Produces ''' instead of the '\''
+    assertThat(chInLiteralBlock.toString()).isEqualTo("'''");
+
+    String str = "\\'";
+    // Works but character is transformed into a String, user would need to do this
+    CodeBlock strInLiteralBlock = CodeBlock.builder().add("'$L'", str).build();
+    assertThat(strInLiteralBlock.toString()).isEqualTo("'\\''");
+
+    char ch2 = '\'';
+    // Util method not public, so user would need to create its own version
+    CodeBlock utilInBlock = CodeBlock.builder().add("'$L'", Util.characterLiteralWithoutSingleQuotes(ch2)).build();
+    assertThat(utilInBlock.toString()).isEqualTo("'\\''");
+  }
+
+  @Test public void singleQuoteCharacterInCharacter() {
+    char c = '\'';
+    CodeBlock block = CodeBlock.builder().add("$C", c).build();
+    assertThat(block.toString()).isEqualTo("'\\''");
+  }
```

Comment:
Perhaps add tests for other escaped literals?

Reply:
Yeah, as mentioned in the description, I was planning to add more tests :). I've just added some more tests for other escaped character literals and squashed the commits.